src/Query/types.ts

```diff
@@ -0,0 +1,48 @@
+import { Filter } from 'mongodb';
+import { Document } from '../SearchIndex/types';
+
+type PathObj = {
+  value: string;
+  multi: string;
+};
+
+type Path = string | PathObj;
+
+export type Score = {
+  boost: {
+    value: number;
+  };
+};
+
+export type Part = {
+  text: {
+    path: Path | Path[];
+    query: string[];
+    score?: Score;
+    synonyms?: string;
+  };
+};
+
+export type CompoundPart = {
+  compound: {
+    must: Part[];
+    mustNot?: Part[];
+    score?: Score;
+  };
+};
+
+type Phrase = {
+  phrase: {
+    path: string | string[];
+    query: string | string[];
+  };
+};
+
+type Equals = {
+  equals: {
+    path: string | string[];
+    value: boolean;
+  };
+};
+
+export type Must = Phrase | Equals | Filter<Document>;
```

Comment:
I included this because [here](https://github.com/mongodb/docs-search-transport/blob/main/src/Query/index.ts#L234) we concatenate `filters` with `must` and `filters` is of type `Filter<Document>[]`.

Reply:
should I change it to `Document`?

Reply:
mongodb.Document makes sense here. and the Filter can still pass 👍